### DISCUSSION TITLE
How to change prompt and keep colors?

### DISCUSSION THREAD

**Discussion by givihuda** [microsoft/WSL#11897] [Q&A] [answered]
After logging into WSL, I get the nice color prompt, but it says my name computer name in it, like this:

`my_first_name@AB1234:~$ `

I found a way to adjust this basically:

`# On /home/wsl/.profile
export PS1="\D{%Y-%m-%d-%I:%M%p} \u@$WSL_DISTRO_NAME: "`
From: https://alexanderallen.medium.com/customizing-my-wsl-bash-prompt-374daf85a7ae

However, it strips the colors and other things so it ends up like this:

`me@Ubuntu: `

It blends in too much with everything else, and doesn't show things like pathing - like `~/.cache$` - and I'm not seeing how to just change the username and computer name but keep the rest of the look/behavior intact.

**Comment by zcobol** [comment, marked as answer]
Look into `.bashrc` file located in your home directory for `PS1` prompt settings:

`PS1='${debian_chroot:+($debian_chroot)}\[\033[01;32m\]\u@\h\[\033[00m\]:\[\033[01;34m\]\w\[\033[00m\]\$ '
`

Look at `\033[01;32m` which is an escape sequence, with `\033[` being CSI (control sequence introducer) and `32` is ANSI green.  There's a good intro at https://en.wikipedia.org/wiki/ANSI_escape_code

Your prompt `PS1="\D{%Y-%m-%d-%I:%M%p} \u@$WSL_DISTRO_NAME: ` has no colors defined.
